### PR TITLE
gnomAD updates

### PR DESCRIPTION
**Pull request by mwalker174**
A number of bug fixes encountered while processing gnomAD-v3, 1KGP, and GMKF. This includes the replacement of tabix with GATK for streaming evidence records from buckets (with suggested changes incorporated from #26). Docker updates forthcoming. Addresses #28.

- Add GATK to sv-base docker; use GATK for PE/SR/RD localization
- fix fragile cleanvcf1 block
- Fix BedCluster too
- Use CHR2 for end locus in svtk count-sr
- Pad RDTest bincov localization; fix inputs for cromwell 51
- Correct bounds for RDTestChromosome
- Update to PrintSVEvidence and start fixing WDLs for cromwell v51
- Fix sv-base dockerfile
- Fix data.table dependency
- Switch to .RD.txt.gz extension for depth evidence files
- Fix localized filenames in 02 TestChromosome tasks
- Fix tabix arguments and a few evidence file names
- random forest fix: select cutoff from AUC and FDR
- fixes to SR_genotype.opt_part2.sh
- bug fixed: 1. runtime_override_collect_vids_per_sample were not passed in Module05_06.wdl; 2. wrong white list were used in RdGenotype
- unnecessary functions removed
- -a added for ConcatVcf
- Module07-09 wdl scripts renamed
- Prevent empty PE and SR common site test shards in 02

### CODE REVIEW

File: wdl/PETestChromosome.wdl
Comment (cwhelan) [inline wdl/PETestChromosome.wdl:112]: Some of these changes might conflict with my proposed changes to support empty shards in https://github.com/broadinstitute/gatk-sv/pull/54 -- just an fyi. It'd be nice if we can make sure the resulting code after merges doesn't error on empty shards.

**Review by cwhelan — COMMENTED** [review]
Here are some comments; I held off on looking at the contents of module07 and 08 since it sounds like that might still be in flux. It'd be good to rename the current `module07.wdl` which is annotations, to `module08.wdl`, maybe in a separate commit?

File: src/sv-pipeline/04_variant_resolution/scripts/clean_vcf_part1.sh
Comment (mwalker174) [reply src/sv-pipeline/04_variant_resolution/scripts/clean_vcf_part1.sh:40]: It turns out that wasn't possible because the `Type` is getting changed from Integer to String.

File: src/svtk/svtk/adjudicate/random_forest.py
Comment (mwalker174) [reply src/svtk/svtk/adjudicate/random_forest.py:123]: Must have been for debugging, so I'll delete

File: src/svtk/svtk/adjudicate/random_forest.py
Comment (mwalker174) [reply src/svtk/svtk/adjudicate/random_forest.py:193]: Changed `learn_cutoff` -> `learn_cutoff_dist` and `learn_cutoff_V2` -> `learn_cutoff_fdr`

File: src/svtk/svtk/cli/pesr_test.py
Comment (mwalker174) [reply src/svtk/svtk/cli/pesr_test.py:314]: Done

File: wdl/AnnotateVcf.wdl
Comment (mwalker174) [reply wdl/AnnotateVcf.wdl:137]: Thanks for pointing this out. It depends on the way the tool is implemented, but I just tested SelectVariants and it does seem to support block compressed output and actually generates an index automatically, so this block of code isn't needed.

File: wdl/PETestChromosome.wdl
Comment (mwalker174) [reply wdl/PETestChromosome.wdl:112]: Yes I'll do a rebase of this branch after yours is merged and make sure the empty shard check is in place

File: wdl/Module07FilterCleanupQualRecalibration.wdl
Comment (mwalker174) [reply wdl/Module07FilterCleanupQualRecalibration.wdl:137]: Done

File: wdl/Module07FilterCleanupQualRecalibration.wdl
Comment (mwalker174) [reply wdl/Module07FilterCleanupQualRecalibration.wdl:197]: Done

**Comment by mwalker174** [comment]
Thanks for the review @cwhelan. Somehow there were a number of duplicated WDLs for modules 07-09, so I cleaned those up. Also updated test jsons for former 07 to be 08. 

I addressed most of your comments, but I still want to hold off on modifying 07 and 09 since I believe they are working as is, and we don't have jsons for testing changes right now. I opened issue #55 to track that we need to review these modules still.

File: wdl/Module07XfBatchEffect.wdl
Comment (cwhelan) [inline wdl/Module07XfBatchEffect.wdl:371]: This for loop seems unnecessary to me.

File: wdl/Module07XfBatchEffect.wdl
Comment (cwhelan) [inline wdl/Module07XfBatchEffect.wdl:369]: This doesn't seem like just a header, it gets pasted to the processed files.. I guess I think of header as always being column headers, maybe calling this row_headers.txt would make this more clear.

File: wdl/Module07XfBatchEffect.wdl
Comment (cwhelan) [inline wdl/Module07XfBatchEffect.wdl:342]: If it works now that's fine, but I feel like this task could be greatly simplified by either using the `join` command or a small python script.

File: wdl/Module07XfBatchEffect.wdl
Comment (cwhelan) [inline wdl/Module07XfBatchEffect.wdl:560]: Are there so many of these files that you can't use `{sep=" " ...}`?

File: wdl/Tasks04.wdl
Comment (cwhelan) [inline wdl/Tasks04.wdl:419]: Is 500 bases of padding always sufficient to get the evidence we needed? 

Should we parameterize this value? I could also see eventually using a value computed from the data (based on insert size metrics distributions, for example).

File: dockerfiles/sv-base/Dockerfile
Comment (cwhelan) [inline dockerfiles/sv-base/Dockerfile:11]: It might be helpful to make a ticket to update to a gatk release rather than this commit once PrintSVEvidence is merged into GATK master.

**Review by cwhelan — APPROVED** [review]
This looks good to me in general. I know you've tested this out on real data so take my comments more as suggestions than requirements to fix. There are some places in the bash tasks that look overly complicated to my eye but perhaps you ran into issues that required the complexity -- it might be good to sprinkle in some comments about why things are implemented the way they are. As noted earlier, I think there are some places where bcftools would do things more efficiently and enforce valid VCF -- perhaps you could expand that ticket to note some of them.

File: README.md
Comment (mwalker174) [reply README.md:80]: samtools doesn't support RP so `CramToBamRequesterPays` doesn't localize the CRAM.

File: inputs/GATKSVPipelineSingleSample.ref_panel_1kg.na12878.json
Comment (mwalker174) [reply inputs/GATKSVPipelineSingleSample.ref_panel_1kg.na12878.json:7]: Oops, I had forgotten to move these. I moved all newly generated files to `gs://gatk-sv-resources-public/hg38/v0/sv-resources/ref-panel/1KG/v2/` and renamed the reference panel from `ref_panel_1kg_v1` to `ref_panel_1kg_v2`. Anything currently in `gs://gcp-public-data--broad-references` I am keeping the same to reduce transfer cost burden on the Methods project, even though these are labeled as `v1`.

File: inputs/GATKSVPipelineSingleSample.ref_panel_1kg.na12878.json
Comment (mwalker174) [reply inputs/GATKSVPipelineSingleSample.ref_panel_1kg.na12878.json:262]: Done

File: wdl/DepthClustering.wdl
Comment (mwalker174) [reply wdl/DepthClustering.wdl:233]: Yes, as far as I know you are only allowed inline `if` statements inside of the command block.

File: wdl/Module07FilterCleanupQualRecalibration.wdl
Comment (mwalker174) [reply wdl/Module07FilterCleanupQualRecalibration.wdl:5]: It's probably because you don't have read permission (I get an error as well).

File: wdl/Module07FilterOutlierSamples.wdl
Comment (mwalker174) [reply wdl/Module07FilterOutlierSamples.wdl:60]: Done

File: wdl/Module07FilterOutlierSamples.wdl
Comment (mwalker174) [reply wdl/Module07FilterOutlierSamples.wdl:185]: Let's roll this into #78, I've updated the title.

File: wdl/Tasks04.wdl
Comment (mwalker174) [reply wdl/Tasks04.wdl:419]: 500 is default the "outer" padding for `svtk count-pe` so hard-coding is okay in this case. 

It would be better to adjust this per-batch based on insert size distribution, but this would take a bit of work to optimize. It's not obvious how to do that given that we don't usually batch based on the mean or variance of the insert size distribution, although they may correlate with dosage score in some way.

File: dockerfiles/sv-base/Dockerfile
Comment (mwalker174) [reply dockerfiles/sv-base/Dockerfile:11]: Created #81

**Comment by mwalker174** [comment]
Thanks @cwhelan! There is quite a bit of work to do on Module07 and so I have renamed #78 more generally. We do need to move ahead with gnomAD ASAP so I am going to punt this down the road and merge.

File: README.md
Comment (cwhelan) [reply README.md:80]: Oh, of course, sorry..

**Pull request merged**
